hasher-matcher-actioner/hmalib/common/config.py

```diff
@@ -0,0 +1,282 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Helpers for writing simple configs backed by a DynamoDB table
+
+Uses dataclass reflection to try and simplifying going between
+AWS API types and local types. There's likely already an existing
+library that exists somewhere that does this much better.
+"""
+from decimal import Decimal
+import functools
+from dataclasses import dataclass, fields, is_dataclass
+import typing as t
+
+import boto3
+from boto3.dynamodb.conditions import Attr
+
+T = t.TypeVar("T")
+TConfig = t.TypeVar("TConfig", bound="HMAConfig")
+
+
+# module level state, set with HMAConfig.initialize()
+_TABLE_NAME = None
+
+
+functools.lru_cache(maxsize=1)
+
+
+def get_dynamodb():
+    """
+    Get the dynamodb resource.
+
+    Putting this at module level causes problems with mocking, so hide in a
+    function. This is only ever used for meta.client, so maybe it would be
+    better to use that. Probably not thread safe.
+    """
+    return boto3.resource("dynamodb")
+
+
+class HMAConfigSerializationError(ValueError):
+    pass
+
+
+@dataclass
+class HMAConfig:
+    """
```

Comment:
Can you add a doc comment or a post_init validation to ensure no one is using ConfigName or ConfigType field names. Extremely unlikely, because python attributes should not be CamelCase, but just in case?

Reply:
Adding doc